### PR TITLE
Telemetry Operator - Add component for creating webhook certificates

### PR DESCRIPTION
**Pull request by lindnerby**
**Description**
Since certificate creation via helm is sometimes causing CR creation to fail for telemetry-operator, read more here https://github.com/kyma-project/kyma/issues/15765, we decide to create an application that creates a certificate and key to be used by a defined validating webhook and gets shared with the webhook server.
Will be used as an init container besides the operator pod and shares the key and cert via emptyDir volume.

It is planned to extend this to be able to configure certificate handling for mutating webhooks and conversion webhooks as well.

Changes proposed in this pull request:

- Add component for creating certificate for validating webhook
- Will write to shared volume dir and set ca bundle at webhook config
- Using self-signed certificate and no "official" ca
- Already adding role for validatingwebhook to read and update
- Some cleanup for role generation and kubebuilder markers in telemetry-operator

**Related issue(s)**
* https://github.com/kyma-project/kyma/issues/15765

### CODE REVIEW

**Comment by pbochynski** [comment]
Hi @lindnerby,
you requested a review from 14 persons, but your PR doesn't look ready to review. Please provide the description, or change it to Draft.

**Comment by pbochynski** [comment]
I think it would be nice if you talk to @muralov and maybe you can align. See another PR:
https://github.com/kyma-project/kyma/pull/15639
I am not talking about sharing code, but at least choosing the same pattern (init container vs job)

**Comment by lindnerby** [comment]
> I think it would be nice if you talk to @muralov and maybe you can align. See another PR: #15639 I am not talking about sharing code, but at least choosing the same pattern (init container vs job)

Nice, thanks for the hint. Will do.

**Review by skhalash — DISMISSED** [review]

File: components/webhook-cert-init/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Webhook Certificate Initialization
+
+## Overview
+
+This application can be used to initialize webhooks with a certificate and key. Best used as a Init Container in Operator Pods where certificate is shared by emptyDir Volume.
```
Comment (grego952) [inline components/webhook-cert-init/README.md:5]:
```suggestion
This application can be used to initialize webhooks with a certificate and key. Best used as an Init Container in the Operator Pods where the certificate is shared by the `emptyDir` Volume.
```

**Review by skhalash — DISMISSED** [review]

**Review by grego952 — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Review by skhalash — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**